src/PrestaShopBundle/Resources/config/services/adapter/services.yml

```diff
@@ -88,3 +91,12 @@ services:
         arguments:
             - "@prestashop.core.admin.data_provider.addons_interface"
             - "@prestashop.adapter.admin.data_provider.module"
+
+    prestashop.adapter.pdf_generator.invoice:
+        class: PrestaShop\PrestaShop\Adapter\Invoice\InvoicePDFGenerator
+        arguments: ['@prestashop.adapter.manager.pdf', '@prestashop.adapter.data_provider.pdf']
+
+    # Data managers
+    prestashop.adapter.manager.pdf:
+        class: PrestaShop\PrestaShop\Adapter\PDF\PDFManager
+        arguments: ["@=service('prestashop.adapter.legacy.context').getSmarty()"]
```

Comment:
Do we really need the one from Context, or do you think it may may work with:

```yaml
prestashop.smarty:
    class: Smarty
```
and then use this service instead of rely on legacy context ?

Reply:
the `prestashop.smarty` doesn't have custom prestashop functions (in this case `{l ...}` function is needed) so it doesn't work

Reply:
aw :(